File: Harvard_CS50_Introduction_to_CS/CS50-Pset6-master/caesar.py
from sys import argv
from sys import exit
from cs50 import get_string

# check the program running with one command-line argument
if len(argv) != 2:
    print("Usage: python caesar.py k")
    exit(1)

# obtain integer k
k = int(argv[1])
# ask user input plaintext
plaintext = get_string("plaintext: ")
print("ciphertext: ", end="")
for char in plaintext:
    if char.isalpha() == True:
        if char.isupper():
            ci_letter = chr((ord(char) + k - 65) % 26 + 65)
            print(ci_letter, end="")
        else:
            ci_letter = chr((ord(char) + k - 97) % 26 + 97)
            print(ci_letter, end="")
    else:
        print(char, end="")
print()
exit(0)

File: CS50_2020/Week2/Caesar/caesar.c
#include<stdio.h>
#include<cs50.h>
#include<string.h>
#include<ctype.h>
#include<stdlib.h>

int main(int argc, string argv[])
{
    // if the there is no second command, then print Usage: ...
    if (argc != 2)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }

    // if there is second input, check if all characters are integer
    // not then print Usage: ... and return 1 (any number ! 0)
    int lenArgv = strlen(argv[1]);
    for (int i = 0; i < lenArgv; i++)
    {
        if (!isdigit(argv[1][i]))
        {
            printf("Usage: ./caesar key\n");
            return 1;
        }
    }

    // convert command-string input into an integer by atoi function
    int k = atoi(argv[1]);

    // prompt user for a string
    string plaintext = get_string("plaintext: ");

    // plaintext length
    int lenPT = strlen(plaintext);

    // print out cipertext after encoded
    printf("ciphertext: ");

    // check if every character in the plaintext is upper, lower or non alphabet
    // encode them by using the key k and ci = (pi + k) % 26 formula
    for (int i = 0; i < lenPT; i++)
    {
        if (islower(plaintext[i]))
        {
            printf("%c", (plaintext[i] + k - 97) % 26 + 97);
        }
        else if (isupper(plaintext[i]))
        {
            printf("%c", (plaintext[i] + k - 65) % 26 + 65);
        }
        else
        {
            printf("%c", plaintext[i]);
        }
    }
    printf("\n");

    // return 0 as the program ran successfully
    return 0;
}